types/src/eds.rs

```diff
@@ -468,6 +468,14 @@ impl From<ExtendedDataSquare> for RawExtendedDataSquare {
     }
 }
 
+impl TryFrom<RawExtendedDataSquare> for ExtendedDataSquare {
+    type Error = Error;
+
+    fn try_from(raw: RawExtendedDataSquare) -> std::result::Result<Self, Self::Error> {
+        ExtendedDataSquare::from_raw(raw, AppVersion::V2)
```

Comment:
I don't think we can assume any particular `AppVersion` here, and this is likely the reason why there is no `Deserialize` derived here nor the `try_from`. I don't have any idea if we could do better there than just require using `ExtendedDataSquare::from_raw`

Reply:
In the [rpc](https://github.com/eigerco/lumina/blob/main/rpc/src/share.rs#L65) we work this around by requiring whole `ExtendedHeader` instead of just height when requesting EDS (and other share related stuff). We made an exclusion for the `RowNamespaceData` as requesting it for parity shares through celestia-node rpc is not supported